src/sentry/static/sentry/app/views/eventsV2/eventDetails/index.tsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {Params} from 'react-router/lib/Router';
+import DocumentTitle from 'react-document-title';
+import {Location} from 'history';
+
+import {t} from 'app/locale';
+import GlobalSelectionHeader from 'app/components/organizations/globalSelectionHeader';
+import {PageContent} from 'app/styles/organization';
+import NoProjectMessage from 'app/components/noProjectMessage';
+import {Organization} from 'app/types';
+import SentryTypes from 'app/sentryTypes';
+import withOrganization from 'app/utils/withOrganization';
+
+import EventView from '../eventView';
+import EventDetailsContent from './content';
+
+type Props = {
+  organization: Organization;
+  location: Location;
+  params: Params;
+};
+
+class EventDetails extends React.Component<Props> {
+  static propTypes: any = {
+    organization: SentryTypes.Organization.isRequired,
+    location: PropTypes.object.isRequired,
+  };
+
+  getEventSlug = (): string => {
+    const {eventSlug} = this.props.params;
+
+    if (typeof eventSlug === 'string') {
+      return eventSlug.trim();
+    }
+
+    return '';
+  };
+
+  getEventView = (): EventView => {
+    const {location} = this.props;
+
+    return EventView.fromLocation(location);
+  };
+
+  getDocumentTitle = (name: string | undefined): Array<string> => {
+    return typeof name === 'string' && String(name).trim().length > 0
+      ? [String(name).trim(), t('Discover')]
+      : [t('Discover')];
+  };
+
+  render() {
+    const {organization, location, params} = this.props;
+    const eventView = this.getEventView();
+
+    const documentTitle = this.getDocumentTitle(eventView.name).join(' - ');
+
+    return (
+      <DocumentTitle title={`${documentTitle} - ${organization.slug} - Sentry`}>
+        <React.Fragment>
+          <GlobalSelectionHeader organization={organization} />
+          <PageContent>
+            <NoProjectMessage organization={organization}>
+              <EventDetailsContent
+                organization={organization}
```

Comment:
Could we pass eventView as a prop to this component instead of reconstructing it from the `location` in EventDetailsContent?

Reply:
Good catch. fixed in https://github.com/getsentry/sentry/pull/15562/commits/4099853ee6a44956d77f4b9fffaaa842b1b58e0f